### PR TITLE
feat: Extend to vmail and solr status checks

### PR DESCRIPTION
**Pull request by kwisatz**
Extends the check script to be able to check solr and vmail status too.
Adds perfdata to results.

Might not be the fanciest python, but it works.
Willing to improve on it though.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,14 +3,19 @@ Plugin for checking a local Mailcow instance via the Mailcow API using Icinga2 /
 # Usage
 
 * **-d / --domain** The domain and optional port to reach the Mailcow instance
+* **-e / --endpoint** The endpoint to check (status/containers, status/solr, status/vmail)
```
Comment (LukasK13) [inline README.md:6]: it might be easier to require containers / solr / vmail without status/. This would also require a slight code change.

**Review by LukasK13 — CHANGES_REQUESTED** [review]
Thank you very much for your pull request.
If you could change the endpoint parameter syntax (see my review) I would love to accept this pull request.

File: README.md (hunk 1 above)
Comment (kwisatz) [reply README.md:6]: I thought about that. But I would like to also query e.g. "mailq/all" which doesn't use the status path.
If we want to allow that, we shouldn't hardcode "status" at this point IMHO.

**Comment by kwisatz** [comment]
Please see my inline comment on why I'm not sure that's practical.

File: README.md (hunk 1 above)
Comment (LukasK13) [reply README.md:6]: In this case, you are absolutely right. Do you intend to add this too at a later point?

**Comment by kwisatz** [comment]
Yes, I do intend to add the mailq check and maybe something more if there's anything else that is useful for checks.

**Comment by LukasK13** [comment]
Awesome. Looking forward to your pull request.
Thanks again.

**Pull request merged**